adobe/sha.hpp

```diff
@@ -239,8 +241,9 @@ std::uint64_t stuff_into_state(typename HashTraits::message_block_type& state,
             v <<= message_end - 8;
 
         *dst |= v;
-        stuff_bit_offset += num_bits;
-        n_stuffed_bits += num_bits;
+        ADOBE_ASSERT(num_bits < 8);
+        stuff_bit_offset += static_cast<std::uint16_t>(num_bits);

```

Comment:
Maybe a comment explaining how you know it won't overflow is warranted.

Reply:
I'm not an expert in this code - it took me a bit to figure out that `num_bits` < 8 here. I'm not certain, but it looks like `stuff_bit_offset` is always used in modulo value_bitsize_k and can warp (unsigned). `n_stuffed_bits` is 64 bits, so it would take a lot to overflow incrementing by `8`. And I _think_ this is going to be bounded by sizeof message_block_type * 8 - so you'll run out of VM space first (if not addressable space).

Reply:
Well, if you can't prove it, there's always `boost::numeric_cast` to maintain safety.

Reply:
That might be very bad if it is expected to wrap.